compiler/rustc_pattern_analysis/src/usefulness.rs

```diff
@@ -734,6 +734,22 @@ struct UsefulnessCtxt<'a, Cx: TypeCx> {
     /// Collect the patterns found useful during usefulness checking. This is used to lint
     /// unreachable (sub)patterns.
     useful_subpatterns: FxHashSet<PatId>,
+    complexity_limit: Option<usize>,
+    complexity_level: usize,
+}
+
+impl<'a, Cx: TypeCx> UsefulnessCtxt<'a, Cx> {
+    fn increase_complexity_level(&mut self, complexity_add: usize) -> Result<(), Cx::Error> {
+        self.complexity_level += complexity_add;
+        if self
+            .complexity_limit
+            .map(|complexity_limit| complexity_limit < self.complexity_level)
+            .unwrap_or(false)
```

Comment:
```suggestion
            .is_some_and(|complexity_limit| complexity_limit < self.complexity_level)
```

Reply:
I'll make a new clippy lint for this if it doesn't exist. :o